dist/tools/picotool/picotool.sh

```diff
@@ -0,0 +1,15 @@
+#!/bin/sh
+
+# $1 contains generated elf file - $FLASHFILE from makefile
+
+ELFFILE="$1"
+UF2FILE="${UF2FILE:-${ELFFILE%%.elf}.uf2}"
+
```

Comment:
For this script it might be good to check the udev rules and to print a message if the udev rules are not appropriate to run `picotool` without root.

https://github.com/raspberrypi/picotool/tree/master?tab=readme-ov-file#linux--macos

Reply:
Otherwise the programming will fail when not running `make flash` as root:

```
/home/chris/RIOTstuff/riot-picotool/RIOT/dist/tools/picotool/picotool.sh /home/chris/RIOTstuff/riot-picotool/RIOT/tests/sys/shell/bin/rpi-pico/tests_shell.elf
[INFO] Converting /home/chris/RIOTstuff/riot-picotool/RIOT/tests/sys/shell/bin/rpi-pico/tests_shell.elf to /home/chris/RIOTstuff/riot-picotool/RIOT/tests/sys/shell/bin/rpi-pico/tests_shell.uf2
[INFO] Loading /home/chris/RIOTstuff/riot-picotool/RIOT/tests/sys/shell/bin/rpi-pico/tests_shell.uf2 onto Pico
No accessible RP-series devices in BOOTSEL mode were found.

but:

RP2040 device at bus 3, address 6 appears to be in BOOTSEL mode, but picotool was unable to connect. Maybe try 'sudo' or check your permissions.
[INFO] Verifying /home/chris/RIOTstuff/riot-picotool/RIOT/tests/sys/shell/bin/rpi-pico/tests_shell.uf2 on Pico
No accessible RP-series devices in BOOTSEL mode were found.

but:

RP2040 device at bus 3, address 6 appears to be in BOOTSEL mode, but picotool was unable to connect. Maybe try 'sudo' or check your permissions.
```

Reply:
```suggestion
udev_warning='[WARNING] Picotool udev rules are not set yet. You won`t be able to '\
'run picotool without sudo. See: '\
'https://github.com/raspberrypi/picotool/tree/master?tab=readme-ov-file#linux--macos'

if ! [ -e /etc/udev/rules.d/99-picotool.rules ]
then
  printf '\033[1;31m%s\033[0m\n' "$udev_warning"
fi
```

Something like this. That will create a colored warning (unfortunately the `${COLOR_RED}` etc macros are only available in the Makefiles) and point to the solution.

![image](https://github.com/user-attachments/assets/720b02bb-8587-486c-a803-774bff489904)